rfcs/034-location_location_location/README.md

```diff
@@ -0,0 +1,520 @@
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works/dutxh9yu?include=items,identifiers">dutxh9yu</a> /
+          <a href="https://search.wellcomelibrary.org/iii/encore/record/C__Rb1077140?lang=eng">b10771402</a>
+        </td>
+      </tr>
+      <tr>
+        <td>wqrfe / Enquiry Line</td>
+        <td>2</td>
+        <td>
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works?items.locations.locationType=wqrfe&include=items%2Cidentifiers">API examples</a> /
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works/en3eymfx?include=items,identifiers">en3eymfx</a> /
+          <a href="https://search.wellcomelibrary.org/iii/encore/record/C__Rb1529652?lang=eng">b15296520</a>
+        </td>
+      </tr>
+      <tr>
+        <td>wghig / Gallery</td>
+        <td>1</td>
+        <td>
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works?items.locations.locationType=wghig&include=items%2Cidentifiers">API examples</a> /
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works/k3ee5bwq?include=items,identifiers">k3ee5bwq</a> /
+          <a href="https://search.wellcomelibrary.org/iii/encore/record/C__Rb2472458?lang=eng">b24724580</a>
+        </td>
+      </tr>
+      <tr>
+        <td>sgmip / sgmip</td>
+        <td>1</td>
+        <td>
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works?items.locations.locationType=sgmip&include=items%2Cidentifiers">API examples</a> /
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works/v5rcgfrg?include=items,identifiers">v5rcgfrg</a> /
+          <a href="https://search.wellcomelibrary.org/iii/encore/record/C__Rb1307568?lang=eng">b13075688</a>
+        </td>
+      </tr>
+      <tr>
+        <td>gblip / 215 Information Point</td>
+        <td>1</td>
+        <td>
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works?items.locations.locationType=gblip&include=items%2Cidentifiers">API examples</a> /
+          <a href="https://api.wellcomecollection.org/catalogue/v2/works/z3hgbrbk?include=items,identifiers">v5rcgfrg</a> /
+          <a href="https://search.wellcomelibrary.org/iii/encore/record/C__Rb1497535?lang=eng">b1497535x</a>
+        </td>
+      </tr>
+    </table>
+
+    </details>
+
+    <details>
+    <summary>Proposed heuristic for matching Sierra holdings locations to controlled types</summary>
+
+    For Sierra holdings, the API only returns a location *code* (e.g. stax) via fixed field 40.
+
+    We will look up the location code to find the location name, then apply a similar rule to that which we apply for items.
+    If the location code or name are unrecognised, we will skip adding a location to the holding.
+
+    This heuristic allows us to create locations in our controlled type for **99.87% of all holdings**.
+
+    The unmapped Sierra holdings locations are as follows:
```

Comment:
I think some of these, like gblipm/wghir are kinds of open shelves with the eg "Reading Room" as the location label.. But I'm sure @amme2 will be able to give you a steer for each of them.

Reply:
It should be fairly easy to add new values to this list as necessary.

Reply:
Sorry, managed to post this in response to the wrong bit of the thread:

> Yes but they're also almost entirely oversights which need clearing up e.g. 'gallery' is the reading room gallery but there's def more than 1 book up there - must be another code in use! If you can send over a list when you get a chance, I'll get them sorted out.
> 
> The bound in / contained in above combination is a whole other kettle of piranhas though, as you no doubt already know.

I don't quite follow your proposed heuristic for dealing with bound in/contained in. I know there's an issue in ordering the response from the Sierra API but stuff isn't typically catalogued in physical arrangement order. Why would it be appropriate to assume every other item has the same shelf location? Maybe I've misunderstood something.

Reply:
(I haven't implemented this heuristic yet. Currently items that say "in above" won't have any location attached.)

This is based on a possibly faulty assumption that "in above" is going to refer to the location of another item attached to the same bib – and if we can unambiguously say what that location would be, we can use it.

Is "in above" going to refer to something completely different?

Possibly an example would be helpful?

---

Suppose we have a bib with items like:

```
i102 Closed stores MS&AB
i101 Bound in above
i104 Closed stores MS&AB
i103 Bound in above
```

but because of problems getting the order from the Sierra API, we get something like:

```
i101 Bound in above
i102 Closed stores MS&AB
i103 Bound in above
i104 Closed stores MS&AB
```

Because `Closed stores MS&AB` is the only non "in above" location in this list, we can assume that's what the "in above" refers to. When we create locations for those items in the Catalogue API, we'll use "Closed stores MS&AB".

On the other hand, suppose we got something like:

```
i101 Bound in above
i102 Closed stores MS&AB
i103 Bound in above
i104 Open shelves
```

There's no good way for us to know whether the "in above" is attached to the Closed stores or Open shelves, so we wouldn't attach any location to items i101 or i103.

Reply:
Ah yes, that makes sense - sorry I'd forgotten you were looking at bibs first (two days on leave...).

Reply:
No worries, always good to double check!